edb/edgeql-parser/src/parser.rs

```diff
@@ -296,12 +332,26 @@ impl<'s> Parser<'s> {
         let mut value = value;
         if let CSTNode::Production(production) = value {
             if let Some(inline_position) = ctx.spec.inlines.get(&production.id) {
+                let inlined_id = production.id;
                 // inline rule found
                 let args = production.args;
                 let span = get_span_of_nodes(args);
 
                 value = args[*inline_position as usize];
 
+                // save inlined id
+                if let CSTNode::Production(new_prod) = &mut value {
```

Comment:
Do we have any deep inlining chains, since this can be quadratic? (Probably our chains are small enough we don't really care?)

Reply:
I don't know, actually, but I can test it.

Reply:
The longest chain found in fast tests is 8. Not critical.